capport_core/src/transform/sql.rs

```diff
@@ -0,0 +1,82 @@
+use std::fmt::Formatter;
+use std::sync::{Arc, RwLock};
+
+use polars_lazy::frame::LazyFrame;
+use polars_sql::SQLContext;
+
+use crate::transform::common::Transform;
+use crate::{
+    pipeline::results::PipelineResults,
+    util::error::{CpError, CpResult},
+};
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+
+pub struct SqlTransform {
+    pub sql: String,
+}
+
+impl SqlTransform {
+    pub const fn keyword() -> &'static str {
+        "sql"
+    }
+    pub fn new(sql: &str) -> SqlTransform {
+        SqlTransform { sql: sql.to_string() }
+    }
+}
+
+impl Transform for SqlTransform {
+    fn run_lazy(&self, curr: LazyFrame, _results: Arc<RwLock<PipelineResults<LazyFrame>>>) -> CpResult<LazyFrame> {
+        let mut context = SQLContext::new();
+
+        context.register("self", curr);
```

Comment:
i think lets register everything in `_results`, and yes also double register the current lazyframe under its actual name (already in results)

can just add a test that tries to write a join to test whether it works